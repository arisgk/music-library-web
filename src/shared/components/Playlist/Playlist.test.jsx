/* global describe, beforeAll, it, expect */
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { createShallow } from '@material-ui/core/test-utils';
import Wrapper, { Playlist } from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Playlist', () => {
  let shallow;
  let songs;
  let onRowButtonClick;

  beforeAll(() => {
    shallow = createShallow({ untilSelector: Playlist });
    songs = [
      {
        id: 1,
        title: 'Animals',
        url: 'https://some-host.com/animals',
        artist: 'Muse',
        album: 'The 2nd Law',
        albumCover: 'https://some-host.com/muse-the-2nd-law.jpg',
        duration: 263,
      },
      {
        id: 2,
        title: 'Coming Back to Life',
        url: 'https://some-host.com/coming-back-to-life',
        artist: 'Pink Floyd',
        album: 'The Division Bell',
        albumCover: 'https://some-host.com/pink-floyd-the-division-bell.jpg',
        duration: 386,
      },
      {
        id: 3,
        title: 'Fade to Black',
        url: 'https://some-host.com/fade-to-black',
        artist: 'Ride the Lightning',
        album: 'Metallica',
        albumCover: 'https://some-host.com/metallica-ride-the-lightning.jpeg',
        duration: 420,
      },
    ];
    onRowButtonClick = () => {};
  });

  it('renders correctly', () => {
    const currentSong = 1;
    const wrapper = shallow(
      <Wrapper
        songs={songs}
        currentSong={currentSong}
        onRowButtonClick={onRowButtonClick}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('updates current song row background', () => {
    let currentSong = 1;
    let wrapper = shallow(
      <Wrapper
        songs={songs}
        currentSong={currentSong}
        onRowButtonClick={onRowButtonClick}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();

    currentSong = 2;
    wrapper = shallow(
      <Wrapper
        songs={songs}
        currentSong={currentSong}
        onRowButtonClick={onRowButtonClick}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('shows play button on hover', () => {
    const currentSong = 1;
    const wrapper = shallow(
      <Wrapper
        songs={songs}
        currentSong={currentSong}
        onRowButtonClick={onRowButtonClick}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.setState({ rowHover: 2 }, () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
