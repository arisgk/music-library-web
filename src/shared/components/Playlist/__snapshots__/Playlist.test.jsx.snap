// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist renders correctly 1`] = `
<WithStyles(Table)>
  <WithStyles(TableHead)>
    <WithStyles(TableRow)>
      <WithStyles(TableCell)
        className="Playlist-head-1"
        padding="checkbox"
        width={64}
      />
      <WithStyles(TableCell)
        className="Playlist-head-1"
      >
        Title
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        className="Playlist-head-1"
      >
        Artist
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        className="Playlist-head-1"
      >
        Album
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </WithStyles(TableHead)>
  <WithStyles(TableBody)>
    <WithStyles(TableRow)
      classes={
        Object {
          "root": "Playlist-currentSongRow-2",
        }
      }
      hover={true}
      key="1"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <WithStyles(TableCell)
        width={64}
      />
      <WithStyles(TableCell)>
        Animals
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Muse
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        The 2nd Law
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
    <WithStyles(TableRow)
      classes={
        Object {
          "root": null,
        }
      }
      hover={true}
      key="2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <WithStyles(TableCell)
        width={64}
      />
      <WithStyles(TableCell)>
        Coming Back to Life
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Pink Floyd
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        The Division Bell
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
    <WithStyles(TableRow)
      classes={
        Object {
          "root": null,
        }
      }
      hover={true}
      key="3"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <WithStyles(TableCell)
        width={64}
      />
      <WithStyles(TableCell)>
        Fade to Black
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Ride the Lightning
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Metallica
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </WithStyles(TableBody)>
</WithStyles(Table)>
`;

exports[`Playlist shows play button on hover 1`] = `
<WithStyles(Table)>
  <WithStyles(TableHead)>
    <WithStyles(TableRow)>
      <WithStyles(TableCell)
        className="Playlist-head-1"
        padding="checkbox"
        width={64}
      />
      <WithStyles(TableCell)
        className="Playlist-head-1"
      >
        Title
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        className="Playlist-head-1"
      >
        Artist
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        className="Playlist-head-1"
      >
        Album
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </WithStyles(TableHead)>
  <WithStyles(TableBody)>
    <WithStyles(TableRow)
      classes={
        Object {
          "root": "Playlist-currentSongRow-2",
        }
      }
      hover={true}
      key="1"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <WithStyles(TableCell)
        width={64}
      />
      <WithStyles(TableCell)>
        Animals
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Muse
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        The 2nd Law
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
    <WithStyles(TableRow)
      classes={
        Object {
          "root": null,
        }
      }
      hover={true}
      key="2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <WithStyles(TableCell)
        width={64}
      />
      <WithStyles(TableCell)>
        Coming Back to Life
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Pink Floyd
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        The Division Bell
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
    <WithStyles(TableRow)
      classes={
        Object {
          "root": null,
        }
      }
      hover={true}
      key="3"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <WithStyles(TableCell)
        width={64}
      />
      <WithStyles(TableCell)>
        Fade to Black
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Ride the Lightning
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Metallica
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </WithStyles(TableBody)>
</WithStyles(Table)>
`;

exports[`Playlist shows play button on hover 2`] = `
<WithStyles(Table)>
  <WithStyles(TableHead)>
    <WithStyles(TableRow)>
      <WithStyles(TableCell)
        className="Playlist-head-1"
        padding="checkbox"
        width={64}
      />
      <WithStyles(TableCell)
        className="Playlist-head-1"
      >
        Title
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        className="Playlist-head-1"
      >
        Artist
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        className="Playlist-head-1"
      >
        Album
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </WithStyles(TableHead)>
  <WithStyles(TableBody)>
    <WithStyles(TableRow)
      classes={
        Object {
          "root": "Playlist-currentSongRow-2",
        }
      }
      hover={true}
      key="1"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <WithStyles(TableCell)
        width={64}
      />
      <WithStyles(TableCell)>
        Animals
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Muse
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        The 2nd Law
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
    <WithStyles(TableRow)
      classes={
        Object {
          "root": null,
        }
      }
      hover={true}
      key="2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <WithStyles(TableCell)
        align="center"
        padding="checkbox"
        width={64}
      >
        <WithStyles(IconButton)
          aria-label="Play Song"
          classes={
            Object {
              "root": "Playlist-iconButtonRoot-3",
            }
          }
          onClick={[Function]}
        >
          <pure(PlayCircleOutlineIcon)
            fontSize="medium"
          />
        </WithStyles(IconButton)>
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Coming Back to Life
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Pink Floyd
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        The Division Bell
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
    <WithStyles(TableRow)
      classes={
        Object {
          "root": null,
        }
      }
      hover={true}
      key="3"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <WithStyles(TableCell)
        width={64}
      />
      <WithStyles(TableCell)>
        Fade to Black
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Ride the Lightning
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Metallica
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </WithStyles(TableBody)>
</WithStyles(Table)>
`;

exports[`Playlist updates current song row background 1`] = `
<WithStyles(Table)>
  <WithStyles(TableHead)>
    <WithStyles(TableRow)>
      <WithStyles(TableCell)
        className="Playlist-head-1"
        padding="checkbox"
        width={64}
      />
      <WithStyles(TableCell)
        className="Playlist-head-1"
      >
        Title
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        className="Playlist-head-1"
      >
        Artist
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        className="Playlist-head-1"
      >
        Album
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </WithStyles(TableHead)>
  <WithStyles(TableBody)>
    <WithStyles(TableRow)
      classes={
        Object {
          "root": "Playlist-currentSongRow-2",
        }
      }
      hover={true}
      key="1"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <WithStyles(TableCell)
        width={64}
      />
      <WithStyles(TableCell)>
        Animals
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Muse
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        The 2nd Law
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
    <WithStyles(TableRow)
      classes={
        Object {
          "root": null,
        }
      }
      hover={true}
      key="2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <WithStyles(TableCell)
        width={64}
      />
      <WithStyles(TableCell)>
        Coming Back to Life
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Pink Floyd
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        The Division Bell
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
    <WithStyles(TableRow)
      classes={
        Object {
          "root": null,
        }
      }
      hover={true}
      key="3"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <WithStyles(TableCell)
        width={64}
      />
      <WithStyles(TableCell)>
        Fade to Black
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Ride the Lightning
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Metallica
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </WithStyles(TableBody)>
</WithStyles(Table)>
`;

exports[`Playlist updates current song row background 2`] = `
<WithStyles(Table)>
  <WithStyles(TableHead)>
    <WithStyles(TableRow)>
      <WithStyles(TableCell)
        className="Playlist-head-1"
        padding="checkbox"
        width={64}
      />
      <WithStyles(TableCell)
        className="Playlist-head-1"
      >
        Title
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        className="Playlist-head-1"
      >
        Artist
      </WithStyles(TableCell)>
      <WithStyles(TableCell)
        className="Playlist-head-1"
      >
        Album
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </WithStyles(TableHead)>
  <WithStyles(TableBody)>
    <WithStyles(TableRow)
      classes={
        Object {
          "root": null,
        }
      }
      hover={true}
      key="1"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <WithStyles(TableCell)
        width={64}
      />
      <WithStyles(TableCell)>
        Animals
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Muse
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        The 2nd Law
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
    <WithStyles(TableRow)
      classes={
        Object {
          "root": "Playlist-currentSongRow-2",
        }
      }
      hover={true}
      key="2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <WithStyles(TableCell)
        width={64}
      />
      <WithStyles(TableCell)>
        Coming Back to Life
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Pink Floyd
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        The Division Bell
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
    <WithStyles(TableRow)
      classes={
        Object {
          "root": null,
        }
      }
      hover={true}
      key="3"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <WithStyles(TableCell)
        width={64}
      />
      <WithStyles(TableCell)>
        Fade to Black
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Ride the Lightning
      </WithStyles(TableCell)>
      <WithStyles(TableCell)>
        Metallica
      </WithStyles(TableCell)>
    </WithStyles(TableRow)>
  </WithStyles(TableBody)>
</WithStyles(Table)>
`;
